### PR TITLE
Applicant resend link to reference

### PR DESCRIPTION
**Pull request by SoLetsDev**
ECER-2113: BE work to resent email to reference

## Description

- Added two new endpoints to resend invite link to reference

## Checklist
- [x] I have tested these changes locally.
- [x] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

### CODE REVIEW

File: src/ECER.Resources.Documents/Applications/ApplicationRepository.cs
Comment (qsl-deepak) [inline src/ECER.Resources.Documents/Applications/ApplicationRepository.cs:1]: do you need this here?: mapper.Map(request, characterReference);

File: src/ECER.Resources.Documents/Applications/IApplicationRepository.cs
Comment (qsl-deepak) [inline src/ECER.Resources.Documents/Applications/IApplicationRepository.cs:1]: body could be removed for public record ResendReferenceInviteRequest(string ReferenceId)

Comment (SoLetsDev) [reply src/ECER.Resources.Documents/Applications/IApplicationRepository.cs:1]: great point done :)

File: src/ECER.Resources.Documents/Applications/ApplicationRepository.cs
Comment (SoLetsDev) [reply src/ECER.Resources.Documents/Applications/ApplicationRepository.cs:1]: I do not...

**Pull request merged**